### PR TITLE
DOC-797: Fix download_stac_asset code snippet example

### PR DESCRIPTION
**Pull request by dustgalactic**
- In `asset-reference.md`, fix example and description.
- In `asset-example.ipynb`, fix cell counts and example.

### CODE REVIEW

**Review by seedlit — APPROVED** [review]

File: examples/asset/asset-example.ipynb [hunk 1]
```diff
@@ -71,7 +71,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 4,
+   "execution_count": 3,
```
Comment (janchrizz) [inline examples/asset/asset-example.ipynb:74]: what is the reason for these changes to execution_count?

Comment (dustgalactic) [reply examples/asset/asset-example.ipynb:74]: When I added the file a while back, I didn't check the counts and they were incorrect.
Because I'm editing this file now, I combined your task and the one from my backlog to fix the counts.

**Review by saad-up42 — APPROVED** [review]
